runners/spark/src/main/java/org/apache/beam/runners/spark/metrics/SparkMetricResults.java

```diff
@@ -0,0 +1,184 @@
+ */
+public class SparkMetricResults extends MetricResults {
+  private static final Logger LOG = LoggerFactory.getLogger(SparkMetricResults.class);
+
+  private final Accumulator<SparkMetricsContainer> metricsAccum;
+
+  public SparkMetricResults(Accumulator<SparkMetricsContainer> metricsAccum) {
+    this.metricsAccum = metricsAccum;
+  }
+
+  @Override
+  public MetricQueryResults queryMetrics(MetricsFilter filter) {
+    return new SparkMetricQueryResults(filter);
+  }
+
+  private class SparkMetricQueryResults implements MetricQueryResults {
+    private final MetricsFilter filter;
+
+    SparkMetricQueryResults(MetricsFilter filter) {
+      this.filter = filter;
+    }
+
+    @Override
+    public Iterable<MetricResult<Long>> counters() {
+      Iterable<CounterAggregator> allCounters =
+          metricsAccum.value().getCounters();
+      Iterable<CounterAggregator> filteredCounters = Iterables.filter(allCounters,
+          new Predicate<CounterAggregator>() {
+            @SuppressWarnings("ConstantConditions")
+            @Override
+            public boolean apply(CounterAggregator metricResult) {
+              return matches(filter, metricResult.getKey());
+            }
+          });
+      return Iterables.transform(filteredCounters,
+          new Function<CounterAggregator, MetricResult<Long>>() {
+            @SuppressWarnings("ConstantConditions")
+            @Nullable
+            @Override
+            public MetricResult<Long> apply(CounterAggregator metricResult) {
+              MetricKey key = metricResult.getKey();
+              return new SparkMetricResult<>(key.metricName(), key.stepName(),
+                  metricResult.getValue());
+            }
+          });
+    }
+
+    @Override
+    public Iterable<MetricResult<DistributionResult>> distributions() {
+      Iterable<DistributionAggregator> allDistributions =
+          metricsAccum.value().getDistributions();
+      Iterable<DistributionAggregator> filteredDistributions =
+          Iterables.filter(allDistributions,
+              new Predicate<DistributionAggregator>() {
+            @SuppressWarnings("ConstantConditions")
+            @Override
+            public boolean apply(DistributionAggregator metricResult) {
+              return matches(filter, metricResult.getKey());
+            }
+          });
+      return Iterables.transform(filteredDistributions,
+          new Function<DistributionAggregator, MetricResult<DistributionResult>>() {
+            @SuppressWarnings("ConstantConditions")
+            @Nullable
+            @Override
+            public MetricResult<DistributionResult>
+            apply(DistributionAggregator metricResult) {
+              MetricKey key = metricResult.getKey();
+              return new SparkMetricResult<>(key.metricName(), key.stepName(),
+                  metricResult.getValue().extractResult());
+            }
+          });
+    }
+
+    private boolean matches(MetricsFilter filter, MetricKey key) {
```

Comment:
@bjchambers The following three methods `matches` `matchesName` and `matchesScope` were copied from the direct runner. Perhaps these can be pulled up to a core module as I believe other runners could benefit from them as well.